h5py/h5p.pyx

```diff
@@ -873,9 +873,9 @@ cdef class PropDCID(PropOCID):
             size = H5Pget_virtual_dsetname(self.id, index, NULL, 0)
             name = <char*>emalloc(size+1)
             try:
-                # TODO check return size
-                H5Pget_virtual_dsetname(self.id, index, name, <size_t>size+1)
-                src_dset_name = bytes(name).decode('utf-8')
+                len = H5Pget_virtual_dsetname(self.id, index, name, <size_t>size+1)
+                if len > 0:
+                    src_dset_name = bytes(name).decode('utf-8')
```

Comment:
What happens when `len == 0`? What's returned?

Reply:
oh, the docs said: `Returns a non-negative value if successful; otherwise returns a negative value.`,  
so, should we do it lile this: `if len >= 0`?

Reply:
The issue is around `src_dset_name` possibly not being defined, and should we have a fallback value or an error or do something else?

Reply:
Yes, we should do something when an error is detected.

Reply:
@aragilar should we tackle it like this: 

https://github.com/HDFGroup/hdf5/blob/e54f5a14d28cf9eb1226ce7feab2e2600d3984b4/java/src/jni/h5pDCPLImp.c#L1286-#L1300

Reply:
Negative (error) return values are already checked and converted to Python exceptions in the autogenerated function wrappers (look in `defs.pyx`). So there's no need to handle them here.

Do we need a special case when there are 0 bytes? It may not make sense to have an empty string, but if that's what HDF5 gives us with no error, we can convert it to an empty Python string without introducing any ambiguity or inaccuracy.

So it's not clear to me what these three TODOs are for. Maybe we can just remove the TODOs and leave the code alone.